packages/build-tools/src/steps/functions/checkout.ts

```diff
@@ -0,0 +1,16 @@
+import { BuildFunction } from '@expo/steps';
+import fs from 'fs-extra';
+
+export function createCheckoutBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'checkout',
+    name: 'Checkout',
+    fn: async (stepsCtx) => {
+      stepsCtx.logger.info('Checking out project directory');
+      await fs.copy(stepsCtx.projectSourceDirectory, stepsCtx.workingDirectory, {
```

Comment:
do we need that project in the old location? copy might be unnecessary expensive, maybe `fs.rename` instead?

if you need .eas directory you can:
- download and unpack project to directory A
- copy `.eas` subdirectory from directory A to some location B
- run steps directory based on location B
- in checkout move entire directory A to location C (primary build workingdir)